calculators/BasicItemNetworthCalculator.js

```diff
@@ -0,0 +1,99 @@
+const { getHypixelItemInformationFromId } = require('../constants/itemsMap');
+const { validRunes } = require('../constants/misc');
+const { ValidationError } = require('../helper/errors');
+const { titleCase } = require('../helper/functions');
+const { getPrices } = require('../helper/prices');
+const networthManager = require('../managers/NetworthManager');
+
+/**
+ * Base class for calculating the networth of a basic item like a sack item or essence
+ */
+class BasicItemNetworthCalculator {
+    /**
+     * Creates a new BasicItemNetworthCalculator
+     * @param {object} itemData The item the networth should be calculated for
+     * @param {string} itemData.id The item's id
+     * @param {number} itemData.amount The amount of the item
+     */
+    constructor({ id, amount }) {
+        this.id = id;
+        this.amount = amount;
+        this.skyblockItem = getHypixelItemInformationFromId(this.id) ?? {};
+        this.itemName = this.skyblockItem?.name || titleCase(this.id).split(' ').reverse().join(' '); //TODO: i think this was only needed for essence?
```

Comment:
todo

Reply:
It's also used for `Beastmaster Crest`, `Griffin Upgrade Stone`, `Wisp Upgrade Stone` & `Exp Boost` (Obtaining their rarity)